### PR TITLE
feat: Add support for coalesced pointer events

### PR DESCRIPTION
**Pull request by denessapi**

### CODE REVIEW

File: test/trap.pointer-events.test.js [hunk 1]
```diff
@@ -20,6 +46,7 @@ jest.mock('../src/constants', () => {
 describe('browser with pointer events', () => {
   beforeAll(() => {
     document.body.innerHTML = initialHtml;
+    window.PointerEvent = PointerEventFake;
```
Comment (gbence) [inline test/trap.pointer-events.test.js:49]: Is this change permanent for the rest of the tests?  Or does the test framework reinitialize the `window` object all the time?

Comment (denessapi) [reply test/trap.pointer-events.test.js:49]: The window object gets reinitialized in the other suites (and so PointerEvent will be undefined) as described in these issues:  
https://github.com/testing-library/dom-testing-library/issues/558 and
https://github.com/jsdom/jsdom/pull/2666

Comment (gbence) [reply test/trap.pointer-events.test.js:49]: Thx!

**Review by gbence — APPROVED** [review]
:+1:

**Pull request merged**